src/openrct2-ui/windows/Error.cpp

```diff
@@ -27,42 +27,89 @@ static rct_widget window_error_widgets[] = {
+        window_error_widgets[WIDX_BACKGROUND].bottom = height;
+
+        widgets = window_error_widgets;
+        error.var_480 = 0;
+        if (!gDisableErrorWindowSound)
+        {
+            OpenRCT2::Audio::Play(OpenRCT2::Audio::SoundId::Error, 0, windowPos.x + (width / 2));
+        }
+    }
 
-/**
- *
- *  rct2: 0x0066792F
- *
- * bx: title
- * dx: message
- */
-rct_window* WindowErrorOpen(StringId title, StringId message, const Formatter& args)
-{
-    auto titlez = format_string(title, args.Data());
-    auto messagez = format_string(message, args.Data());
-    return WindowErrorOpen(titlez, messagez);
-}
+    void OnDraw(rct_drawpixelinfo& dpi) override
+    {
+        ScreenCoordsXY leftTop{ windowPos };
+        ScreenCoordsXY rightBottom{ windowPos + ScreenCoordsXY{ width - 1, height - 1 } };
+        ScreenCoordsXY leftBottom{ leftTop.x, rightBottom.y };
+        ScreenCoordsXY rightTop{ rightBottom.x, leftTop.y };
+
+        gfx_filter_rect(
+            &dpi, ScreenRect{ leftTop + ScreenCoordsXY{ 1, 1 }, rightBottom - ScreenCoordsXY{ 1, 1 } },
+            FilterPaletteID::Palette45);
+        gfx_filter_rect(&dpi, ScreenRect{ leftTop, rightBottom }, FilterPaletteID::PaletteGlassSaturatedRed);
+
+        gfx_filter_rect(
+            &dpi, ScreenRect{ leftTop + ScreenCoordsXY{ 0, 2 }, leftBottom - ScreenCoordsXY{ 0, 2 } },
+            FilterPaletteID::PaletteDarken3);
+        gfx_filter_rect(
+            &dpi, ScreenRect{ rightTop + ScreenCoordsXY{ 0, 2 }, rightBottom - ScreenCoordsXY{ 0, 2 } },
+            FilterPaletteID::PaletteDarken3);
+        gfx_filter_rect(
+            &dpi, ScreenRect{ leftBottom + ScreenCoordsXY{ 2, 0 }, rightBottom - ScreenCoordsXY{ 2, 0 } },
+            FilterPaletteID::PaletteDarken3);
+        gfx_filter_rect(
+            &dpi, ScreenRect{ leftTop + ScreenCoordsXY{ 2, 0 }, rightTop - ScreenCoordsXY{ 2, 0 } },
+            FilterPaletteID::PaletteDarken3);
+
+        gfx_filter_rect(
+            &dpi, ScreenRect{ rightTop + ScreenCoordsXY{ 1, 1 }, rightTop + ScreenCoordsXY{ 1, 1 } },
+            FilterPaletteID::PaletteDarken3);
+        gfx_filter_rect(
+            &dpi, ScreenRect{ rightTop + ScreenCoordsXY{ -1, 1 }, rightTop + ScreenCoordsXY{ -1, 1 } },
+            FilterPaletteID::PaletteDarken3);
+        gfx_filter_rect(
+            &dpi, ScreenRect{ leftBottom + ScreenCoordsXY{ 1, -1 }, leftBottom + ScreenCoordsXY{ 1, -1 } },
+            FilterPaletteID::PaletteDarken3);
+        gfx_filter_rect(
+            &dpi, ScreenRect{ rightBottom - ScreenCoordsXY{ 1, 1 }, rightBottom - ScreenCoordsXY{ 1, 1 } },
+            FilterPaletteID::PaletteDarken3);
+
+        draw_string_centred_raw(
+            &dpi, { leftTop + ScreenCoordsXY{ (width + 1) / 2 - 1, 1 } }, _numLines, _text.data(), FontStyle::Medium);
+    }
+
+    void OnUnknown5() override
+    {
+        error.var_480++;
+        if (error.var_480 >= 8)
+            Close();
```

Comment:
```suggestion
        if (error.var_480 >= 8)
        {
            Close();
        }
```
I know it wasn't like this, but we can take a chance to add safety :P